modules/modal/react/lib/ModalContent.tsx

```diff
@@ -212,9 +190,8 @@ const ModalContent = ({
   }, []);
 
   const content = (
-    <Container onClick={handleOutsideClick} {...elemProps}>
+    <Container {...elemProps} ref={modalRef} onClick={onOverlayClick}>
```

Comment:
Change the element of `modalRef`. This is because the PopupStack sets the z-index of the ref that is handed to it. If that was the Modal and not the overlay, the overlay could appear behind content it is supposed to cover.